associated-token-account/program/src/lib.rs

```diff
@@ -70,20 +67,9 @@ pub fn create_associated_token_account(
     wallet_address: &Pubkey,
     spl_token_mint_address: &Pubkey,
 ) -> Instruction {
-    let associated_account_address =
-        get_associated_token_address(wallet_address, spl_token_mint_address);
-
-    Instruction {
-        program_id: id(),
-        accounts: vec![
-            AccountMeta::new(*funding_address, true),
-            AccountMeta::new(associated_account_address, false),
-            AccountMeta::new_readonly(*wallet_address, false),
-            AccountMeta::new_readonly(*spl_token_mint_address, false),
-            AccountMeta::new_readonly(solana_program::system_program::id(), false),
-            AccountMeta::new_readonly(spl_token::id(), false),
-            AccountMeta::new_readonly(sysvar::rent::id(), false),
-        ],
-        data: vec![],
-    }
+    instruction::create_associated_token_account(
```

Comment:
For consistency reasons with other SPL programs I added the new instruction creator to `instruction` module and this legacy function only redirects to it. All the new instructions would be there as well.

Do we want to keep this legacy function for backward compatibility or we should bump the ATA program version to indicate breaking changes and remove it?

Reply:
We need to be fully backwards compatible.  Bumping the program version is going to be too hard to roll out across the ecosystem

Reply:
Sure, all the changes are backward compatible.

When you say bumping the program version is going to be a problem do you mean we should just avoid breaking changes  or we also should not touch the actual version number?

Reply:
Sorry let's keep the legacy function for now, to avoid the need to bump the major version of the crate

Reply:
ok

Reply:
We can at least mark it:
```
#[deprecated(since="1.1.0", note="please use `instruction::create_associated_token_account` instead")]
```
and at least bump this to 1.1

Reply:
That's cool. Done